src/mpid/ch4/shm/posix/release_gather/release_gather_types.h

```diff
@@ -28,6 +28,7 @@ typedef struct MPIDI_POSIX_release_gather_comm_t {
     MPIR_Treealgo_tree_t bcast_tree, reduce_tree;
     int flags_shm_size;
     uint64_t gather_state, release_state;
+    uint64_t alloc_shm_size;
```

Comment:
Why this is needed?

Reply:
You need to record the amount of shm memory allocated, so you should decrease this amount when freeing comm obj.

Reply:
I see. Another question, how is `MPIDI_POSIX_shm_limit_counter` being used? What is manifestation of the bug?

Reply:
Can you please add a brief description for the new `alloc_shm_size` field added into `struct MPIDI_POSIX_release_gather_comm_t`?